### PR TITLE
sigal: 2.3 -> 2.4

### PR DESCRIPTION
**Pull request by ToxicFrog**
This fixes the broken build on nixpkgs 23.11/unstable, and we can drop the permissions patch because it's included upstream.

## Description of changes

Full changelog [here](http://sigal.saimon.org/en/latest/changelog.html). Mostly bugfixes, shouldn't break anything. Requires Pillow ≥ 8, which shouldn't be a problem here.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by matthiasbeyer — APPROVED** [review]

**Comment by emilylange** [comment]
cc @NixOS/darwin-maintainers for the failing darwin builds

**Comment by n8henrie** [comment]
I get a segfault manually running the ffmpeg command from the test:

```console
$ nix run nixpkgs#ffmpeg -- \
    -i "tests/sample/pictures/video/example video.ogv" \
    -y -crf 23 -strict -2 example\ video.mp4
ffmpeg version 6.0 Copyright (c) 2000-2023 the FFmpeg developers
  built with clang version 16.0.6
  configuration: --disable-static --prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0 --target_os=darwin --arch=aarch64 --pkg-config=pkg-config --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --disable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --datadir=/nix/store/xnp8yg516s5x31jnkzw2aw4nxhkdqyhv-ffmpeg-6.0-data/share/ffmpeg --enable-ffmpeg --disable-ffplay --enable-ffprobe --bindir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0-bin/bin --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avutil --enable-postproc --enable-swresample --enable-swscale --libdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0-lib/lib --incdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0-dev/include --enable-doc --enable-htmlpages --enable-manpages --mandir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0-man/share/man --enable-podpages --enable-txtpages --docdir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-ffmpeg-6.0-doc/share/doc/ffmpeg --disable-alsa --enable-bzlib --disable-libcelt --disable-cuda --disable-cuda-llvm --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --disable-frei0r --disable-libfribidi --disable-libgme --enable-gnutls --disable-libgsm --disable-ladspa --enable-libmp3lame --disable-libaom --enable-libass --disable-libbluray --disable-libbs2b --disable-libdc1394 --disable-libdrm --enable-iconv --disable-libjack --disable-libmfx --disable-libmodplug --disable-libmysofa --enable-libopus --disable-librsvg --enable-libsrt --enable-libssh --disable-libtensorflow --enable-libtheora --disable-libv4l2 --disable-v4l2-m2m --disable-vaapi --enable-vdpau --enable-libvorbis --disable-libvmaf --enable-libvpx --disable-libwebp --disable-xlib --disable-libxcb --disable-libxcb-shm --disable-libxcb-xfixes --disable-libxcb-shape --disable-libxml2 --enable-lzma --disable-cuvid --disable-nvdec --disable-nvenc --disable-openal --disable-opencl --disable-libopencore-amrnb --disable-opengl --disable-libopenh264 --disable-libopenjpeg --disable-libopenmpt --disable-libpulse --disable-librav1e --disable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --disable-libvidstab --disable-libvo-amrwbenc --enable-libx264 --enable-libx265 --disable-libxavs --enable-libxvid --disable-libzmq --enable-libzimg --enable-zlib --disable-vulkan --disable-libglslang --disable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping --cc=clang --cxx=clang++
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Input #0, ogg, from 'tests/sample/pictures/video/example video.ogv':
  Duration: 00:00:10.01, start: 0.000000, bitrate: 309 kb/s
  Stream #0:0: Video: theora, yuv420p(bt470bg/bt470m/bt709), 240x98 [SAR 1:1 DAR 120:49], 29.83 fps, 29.87 tbr, 29.87 tbn
    Metadata:
      ORGANIZATION    : ConversorXO
      encoder         : Lavc60.3.100 libtheora
  Stream #0:1: Audio: vorbis, 44100 Hz, stereo, fltp, 112 kb/s
    Metadata:
      encoder         : Lavc60.3.100 libvorbis
Stream mapping:
  Stream #0:0 -> #0:0 (theora (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (vorbis (native) -> aac (native))
Press [q] to stop, [?] for help
[libx264 @ 0x154e07ca0] using SAR=1/1
[libx264 @ 0x154e07ca0] using cpu capabilities: ARMv8 NEON
[libx264 @ 0x154e07ca0] profile High, level 1.2, 4:2:0, 8-bit
[libx264 @ 0x154e07ca0] 264 - core 163 - H.264/MPEG-4 AVC codec - Copyleft 2003-2021 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=3 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=25 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
Output #0, mp4, to 'example video.mp4':
  Metadata:
    encoder         : Lavf60.3.100
  Stream #0:0: Video: h264 (avc1 / 0x31637661), yuv420p(bt470bg/bt470m/bt709, progressive), 240x98 [SAR 1:1 DAR 120:49], q=2-31, 29.87 fps, 29873 tbn
    Metadata:
      ORGANIZATION    : ConversorXO
      encoder         : Lavc60.3.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: N/A
  Stream #0:1: Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 128 kb/s
    Metadata:
      encoder         : Lavc60.3.100 aac
Segmentation fault: 110.0 size=       0kB time=-577014:32:22.77 bitrate=  -0.0kbits/s speed=N/A
$ echo $?
139
```

```console
$ nix-info -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(n8henrie): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/xl2wzz8plgylz5rqzhmisgy8hqvr50sz-source`
```

**Comment by n8henrie** [comment]
Succeeds with ffmpeg from 23.05:

```console
$ nix run github:nixos/nixpkgs/release-23.05#ffmpeg -- \
    -i <(curl -sL 'https://github.com/saimn/sigal/raw/main/tests/sample/pictures/video/example%20video.ogv') \
    -y -crf 23 -strict -2 example\ video.mp4
...
$ echo $?
0
```

**Comment by reckenrode** [comment]
Probably https://github.com/NixOS/nixpkgs/pull/271459, which still hasn’t hit master. Try staging-next and see if it crashes there.

**Comment by n8henrie** [comment]
@reckenrode ffmpeg has a build failure on staging-next:

```console
$ nix build github:nixos/nixpkgs/staging-next#ffmpeg
error: builder for '/nix/store/jcqas40djh7da6y9xrvwcn8ww3jndaf6-python3.11-lxml-4.9.3-3.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/vpvhjfv5090jwdbzddw4p05i9p2j7ys2-libxml2-2.12.3-dev/include/libxml2/libxml/xpath.h:562:1: note: 'xmlXPathInit' has been explicitly marked deprecated here
       > XML_DEPRECATED
       > ^
       > /nix/store/vpvhjfv5090jwdbzddw4p05i9p2j7ys2-libxml2-2.12.3-dev/include/libxml2/libxml/xmlversion.h:447:43: note: expanded from macro 'XML_DEPRECATED'
       > #    define XML_DEPRECATED __attribute__((deprecated))
       >                                           ^
       > 1 warning generated.
       > cc private/tmp/nix-build-python3.11-lxml-4.9.3-3.drv-0/xmlXPathInitpqduk5bi.o -L/nix/store/a0sdv3sgr8pkcnlrsxcr2iz2hh50zg6r-libxml2-2.12.3/lib -L/nix/store/f1az15zczk8cy556gngdgsyigdx591c6-zlib-1.3/lib -L/nix/store/pmyzvssjl6fw5b7lkdvkhfp1q18jjn10-libxslt-1.1.39/lib -lxml2 -o a.out
       > error: command '/nix/store/z4838r114yg3krnvk2jkg88yn4djp9s7-Toolchains/XcodeDefault.xctoolchain/bin/clang' failed with exit code 1
       > /nix/store/d26cnzf9z308c2myx2kym66cxmiyp9vy-stdenv-darwin/setup: line 1566: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/jcqas40djh7da6y9xrvwcn8ww3jndaf6-python3.11-lxml-4.9.3-3.drv'.
error: 1 dependencies of derivation '/nix/store/afzrm8ilq14r12cfky4cdi1l93wjz6av-gtk-doc-1.33.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dbbg2qfaqaki39zprd7k6f46j6lxgz45-glib-2.78.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pxrxwfs625qq3dc2izyki3gyircs8dpv-harfbuzz-7.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w6dazj7vif440cyrms1qa5dv9m2kc80s-fontforge-20230101.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hn5251hn9jyp5xbkczbjy9zc9893b6sw-libass-0.17.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lxqjiyacfacqkmxg4nimn9gm4sljh72d-ffmpeg-6.1.drv' failed to build
```

File: pkgs/applications/misc/sigal/default.nix [hunk 1]
```diff
@@ -7,16 +7,14 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "sigal";
-  version = "2.3";
-  format = "setuptools";
+  version = "2.4";
+  format = "pyproject";
```
Comment (dotlambda) [inline pkgs/applications/misc/sigal/default.nix:11]:
```suggestion
  pyproject = true;
```

File: pkgs/applications/misc/sigal/default.nix [hunk 2]
```diff
@@ -33,6 +31,7 @@ python3.pkgs.buildPythonApplication rec {
     cryptography
 
     setuptools # needs pkg_resources
+    setuptools-scm
```
Comment (dotlambda) [inline pkgs/applications/misc/sigal/default.nix:34]: should be in `nativeBuildInputs`

File: pkgs/applications/misc/sigal/default.nix [hunk 3]
```diff
@@ -33,6 +31,7 @@ python3.pkgs.buildPythonApplication rec {
     cryptography
 
     setuptools # needs pkg_resources
```
Comment (dotlambda) [inline pkgs/applications/misc/sigal/default.nix:33]: https://github.com/saimn/sigal/commit/b45d018cead4238750aa11a84cedc3fd4c8450c7

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Review by dotlambda — APPROVED** [review]

**Pull request merged**